dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -510,6 +543,10 @@ private ContextVariables GetNextStepVariables(ContextVariables variables, Plan s
             {
                 stepVariables.Set(item.Key, value);
             }
+            else if (!string.IsNullOrEmpty(expandedValue) && !stepVariables.Get(item.Key, out var _))
+            {
+                stepVariables.Set(item.Key, expandedValue);
```

Comment:
ContextVariables.Set will check for null, and remove the arg if it's null. You may not need this last conditional (i.e. it could just be "else") if you're okay letting it remove the entry at item.Key.